pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -1168,12 +1168,18 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 		}
 	}
 	// append skipped volumes
+	failedVolCount := 0
 	backup.Status.Volumes = append(backup.Status.Volumes, skipVolInfo...)
 	for _, vol := range backup.Status.Volumes {
 		if vol.Status == stork_api.ApplicationBackupStatusFailed {
-			backup.Status.FailedVolCount++
+			failedVolCount++
 		}
 	}
+	backup.Status.FailedVolCount = failedVolCount
+	err = a.client.Update(context.TODO(), backup)
+	if err != nil {
```

Comment:
Instead of doing it here update the failedVolcount in the retry logic, So that even if it fails, we can take care with retry.